tests/integ/modin/index/test_unique.py

```diff
@@ -33,7 +33,36 @@ def test_index_unique_nan_vs_none_negative():
 
 
 @pytest.mark.parametrize("native_index", NATIVE_INDEX_UNIQUE_TEST_DATA)
+@pytest.mark.parametrize("dropna", [True, False])
 @sql_count_checker(query_count=1)
-def test_index_nunique(native_index):
+def test_index_nunique(native_index, dropna):
     snow_index = pd.Index(native_index)
-    assert snow_index.nunique() == native_index.nunique()
+    assert snow_index.nunique(dropna=dropna) == native_index.nunique(dropna=dropna)
+
+
+def test_index_unique_data_columns_should_not_affect_index_column():
+    native_df = native_pd.DataFrame(
+        {
+            "B": [10, 20, 30, 40, 50],
+            "A": [11, 22, 33, 44, 55],
+            "C": [60, 70, 80, 90, 100],
+        },
+        index=native_pd.Index([5, 4, 3, 2, 1], name="A"),
+    )
+    snow_df = pd.DataFrame(native_df)
+
+    with pytest.raises(AssertionError):
```

Comment:
it's clearer to move this context manager around the `assert_index_equal` alone.